backend/assignments/2/src/main/java/com/kdu/caching/service/ReverseGeocodeAPIService.java

```diff
@@ -0,0 +1,89 @@
+package com.kdu.caching.service;
+
+import com.kdu.caching.constants.Constants;
+import com.kdu.caching.entities.GeocodeData;
+import com.kdu.caching.entities.ReverseGeocodeData;
+import com.kdu.caching.exception.customexceptions.EmptyJsonException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.coyote.BadRequestException;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Slf4j
+@Service
+public class ReverseGeocodeAPIService {
+    @Value("${api-key}")
+    private String apiKey;
+
+    @Value("${reverse-geocoding-url}")
+    private String apiUrl;
+
+    /**
+     * @param latitude Latitude of a location
+     * @param longitude Longitude of a location
+     * @return ReverseGeocodeData
+     * @throws BadRequestException
+     * @throws EmptyJsonException
+     */
+    @Cacheable(value = "reverse-geocoding", key = "#latitude + ',' + #longitude", unless = "#result != null && #result.region.toLowerCase() == 'goa'")
+    public ReverseGeocodeData fetchData(Double latitude, Double longitude) throws BadRequestException, EmptyJsonException {
+        String apiResponse;
+        try{
+        String url = apiUrl + "?access_key=" + apiKey + "&query=" + latitude + "," + longitude;
+        RestTemplate restTemplate = new RestTemplate();
+        apiResponse = restTemplate.getForObject(url, String.class);
+        }
+        catch (Exception e) {
+            log.error("Exception occurred while getting API response");
+            throw new BadRequestException(Constants.INVALID_PARAMETER_MESSAGE);
+        }
+        log.info("API response retrieved successfully");
+        return extractRequiredData(apiResponse);
+    }
+
+    @CacheEvict(value = "reverse-geocoding", key = "#latitude + ',' + #longitude")
```

Comment:
How's this supporting for eviction?

Reply:
not supporting, implemented using some other way.
will take care of removing unnecessary code from next PR